src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -49,6 +51,11 @@ internal partial class CodeFixService : ForegroundThreadAffinitizedObject, ICode
 
         private ImmutableDictionary<object, FixAllProviderInfo> _fixAllProviderMap;
 
+        /// <summary>
+        /// Cache from <see cref="CodeFixProvider"/> instance to its type name.
+        /// </summary>
+        private readonly ConditionalWeakTable<CodeFixProvider, string> _fixerTypeNameCache;
```

Comment:
why cache here?  don't we own CodeFixProvider?  why not have a lazy property on it where it computes and caches this itself?